matrix_sdk_base/src/models/room.rs

```diff
@@ -257,6 +265,63 @@ impl Room {
 
         let member = RoomMember::new(event);
 
+        // find all users that share the same display name as the joining user
+        let users_with_same_name: Vec<_> = self
+            .display_names
+            .iter()
+            .filter(|(_, v)| {
+                member
+                    .display_name
+                    .as_ref()
+                    .map(|n| &n == v)
+                    .unwrap_or(false)
+            })
+            .map(|(k, _)| k)
+            .cloned()
+            .collect();
+
+        // if there is no other user with the same display name -> just use the display name
+        if users_with_same_name.is_empty() {
```

Comment:
Can't we store just the users that have clashes? That way you wouldn't need to duplicate the display names of every user in the room and will get rid of the majority of clones that are happening.

Reply:
That's a good idea. Also I just realised, I forgot to cleanup the names each time a user leaves the room.

Reply:
Also if a user changes their display name.